dot/parachain/network-bridge/wire_message.go

```diff
@@ -0,0 +1,99 @@
+package networkbridge
+
+import (
+	"fmt"
+
+	"github.com/ChainSafe/gossamer/dot/network"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/pkg/scale"
+)
+
+type WireMessage struct {
+	inner any
+}
+
+type WireMessageValues interface {
+	ViewUpdate | ProtocolMessage
+}
+
+func setMyVaryingDataType[Value WireMessageValues](mvdt *WireMessage, value Value) {
+	mvdt.inner = value
+}
+
+func (mvdt *WireMessage) SetValue(value any) (err error) {
+	switch value := value.(type) {
+	case ViewUpdate:
+		setMyVaryingDataType(mvdt, value)
+		return
+	case ProtocolMessage:
+		setMyVaryingDataType(mvdt, value)
+		return
+	default:
+		return fmt.Errorf("unsupported type")
+	}
+}
+
+func (mvdt WireMessage) IndexValue() (index uint, value any, err error) {
+	switch mvdt.inner.(type) {
+	case ProtocolMessage:
+		return 1, mvdt.inner, nil
+	case ViewUpdate:
+		return 2, mvdt.inner, nil
+
+	}
+	return 0, nil, scale.ErrUnsupportedVaryingDataTypeValue
+}
+
+func (mvdt WireMessage) Value() (value any, err error) {
+	_, value, err = mvdt.IndexValue()
+	return
+}
+
+func (mvdt WireMessage) ValueAt(index uint) (value any, err error) {
+	switch index {
+	case 1:
+		return ProtocolMessage{}, nil
+	case 2:
+		return ViewUpdate{}, nil
+	}
+	return nil, scale.ErrUnknownVaryingDataTypeValue
+}
+
+func (w WireMessage) Type() network.MessageType {
+	// TODO: create a wire message type and return that
```

Comment:
Do you plan to address this todo in this PR?

Reply:
created https://github.com/ChainSafe/gossamer/issues/4108